tutorials/multicore/mt301_fillHistAutoBin.C

```diff
@@ -0,0 +1,95 @@
+/// \file
+/// \ingroup tutorial_multicore
+/// Fill histograms w/o in parallel with automatic binning.
+/// Illustrates range synchronization
+///
+/// \macro_code
+///
+/// \author Gerardo Ganis
+/// \date August 2017
+#include "TH1D.h"
+#include "TH2D.h"
+#include "TH3D.h"
+#include "TList.h"
+#include "TRandom3.h"
+#include "TDirectory.h"
+#include "TROOT.h"
+#include "TCanvas.h"
+#include <thread>
+#include <iostream>
```

Comment:
Just a minor comment: We could perhaps try to put stl headers at the end of ROOT includes in order not to mask defects in our headers.

Reply:
Sure! Done.